test/pageobjects/CheckoutPage.js

```diff
@@ -23,22 +27,94 @@ class CheckoutPage extends PageObject {
     }
 
-    get btnCancel() {
-        return browser.$("//button[@id='cancel']");
-    }
-
     get btnContinue() {
         return browser.$("//input[@id='continue']");
     }
 
-    async fillForm(firstName, lastName, zipCodeText) {
+    async completeCheckoutStepOne(firstName, lastName, zipCodeText) {
         await this.firstNameText.setValue(firstName)
         await this.lastNameText.setValue(lastName)
         await this.zipCodeText.setValue(zipCodeText)
         await this.btnContinue.click() }
 
     // checkout step two
 
+    get summaryInfoText() {
+        return browser.$("//div[@class='summary_info']");
+    }
+
+    get btnFinish() {
+        return browser.$("//button[@id='finish']");
+    }
+
+    get itemTotalPriceText() {
+        return browser.$("//div[@class='summary_subtotal_label']");
+    }
+    get taxPriceText() {
+        return browser.$("//div[@class='summary_tax_label']");
+    }
+    get totalPriceText() {
+        return browser.$("//div[@class='summary_total_label']");
+    }
+
+    get allItemsName() {
+        return browser.$$("//div[@class='inventory_item_name']");
+    }
+
+    get allItemsPrices() {
+        return browser.$$("//div[@class = 'inventory_item_price']");
+    }
+
+    get allItemsDescriptions() {
+        return browser.$$('//div[@class="inventory_item_desc"]');
+    }
+
+    async addProductAndProceedToCheckout(option) {
+        switch(option){
+            case 'oneItem':
+                await ProductsPage.btnAddBackpack.click()
+                break
+            case 'multipleItems':
+                await ProductsPage.addMultipleItems()
+                break
+        }
+        await ProductsPage.shoppingCartIcon.click()
+        await CartPage.btnCheckout.click()
+    }
+
+    async checkoutProductData(data) {
+        switch(data){
+            case 'names':
+                let productsNamesText = []
+                const productNames = await this.allItemsName;
+                for (let i = 0; i < productNames.length; i++) {
+                    productsNamesText.push(await productNames[i].getText())
+                }
+                return productsNamesText
+            case 'prices':
+                let productsPrices = [];
+                const productPrice = await this.allItemsPrices;
+                let numberPrice
+                for (let i = 0; i < productPrice.length; i++) {
+                    numberPrice = await productPrice[i].getText()
+                    productsPrices.push(numberPrice.replace("$", ""));
+                }
+                return productsPrices
+            case 'descriptions':
+                let productsDetailsText = [];
+                const productDetail = await this.allItemsDescriptions;
+                for (let i = 0; i < productDetail.length; i++) {
+                    productsDetailsText.push(await productDetail[i].getText());
+                }
+                return productsDetailsText
+        }
```

Comment:
same as above

Reply:
a new method "extractTextData" has been created inside the PageObject file.